myapp/static/js/myapp/base.js

```diff
@@ -0,0 +1,192 @@
+                            var time = document.createTextNode(parseDateTime(responseDict[key]['time']));
+
+                            link.appendChild(div);
+                            div.appendChild(header);
+                            header.appendChild(title);
+                            div.appendChild(body);
+                            body.appendChild(description);
+                            div.appendChild(footer);
+                            footer.appendChild(time);
+
+                            nContainer.appendChild(link);
+                            link.onclick = function() {markAsRead(String(key));};
+                        }
+                    }
+                }
+            }
+        }
+    };
+    xhr.send();
+}
+
+
+function markAsRead(pk=""){
+    var xhr = new XMLHttpRequest();
+    if (pk === "") {
+        var url = myapp.URLS.notificationsAllRead; 
+    }
+    else {
+        var url = getAbsoluteURL(`/notifications/${pk}/nread/`);
+    }
+    xhr.open("POST", url, true);
+
+    xhr.onreadystatechange = function () {
+        if (xhr.readyState === XMLHttpRequest.DONE) {
+            var status = xhr.status;
+            if (status === 0 || (status >= 200 && status < 400)) {
+                console.log("all read");
+            }
+        }
+    };
+    xhr.setRequestHeader('X-CSRFToken', getCookie('csrftoken'));
+    xhr.send();
+    getUnreadCount();
+}
+
+
+function deleteNotifications(pk=""){
+    var xhr = new XMLHttpRequest();
+    if (pk === "") {
+        var url = myapp.URLS.notificationsAllClear;
+    }
+    else {
+        var url = getAbsoluteURL(`/notifications/${pk}/nremove/`);
+    }
+
+    xhr.open("POST", url, true);
+
+    xhr.onreadystatechange = function () {
+        if (xhr.readyState === XMLHttpRequest.DONE) {
+            var status = xhr.status;
+            if (status === 0 || (status >= 200 && status < 400)) {
+                console.log("all deleted");
+            }
+        }
+    };
+    xhr.setRequestHeader('X-CSRFToken', getCookie('csrftoken'));
+    xhr.send();
+    getUnreadCount();
+}
+
+
+function parseDateTime(isoFormat) {
+    date = new Date(isoFormat);
+    return date.toDateString() + " at " + date.toLocaleTimeString().substr(0, 5);
+}
+
+
+function getAbsoluteURL(path){
+    return window.location.protocol + "//" + window.location.host + path;
+}
+
+
+const makeSentenceCase = (s) => {
+    if (typeof s !== 'string') return ''
+    return s.charAt(0).toUpperCase() + s.slice(1);
+}
+
+
+$("#notificationsDropdown").on("show.bs.dropdown", function() {
+    getNotifications();
+    getUnreadCount();
+});
+
+
+$('.dropdown').on('show.bs.dropdown', function() {
+    $(this).find('.dropdown-menu').first().stop(true, true).slideDown(100);
+});
+
+
+$('.dropdown').on('hide.bs.dropdown', function() {
+    $(this).find('.dropdown-menu').first().stop(true, true).slideUp(100);
+});
+
+
+window.setInterval(getUnreadCount, 30000);
```

Comment:
good one, that's an alternative solution for the real-time connection between the browser and the server